notify/teams.py

```diff
@@ -81,18 +140,142 @@ def create_buttons(self, buttons: dict) -> None:
             Adds the button(s) to the teams message
 
         """
-
+        button_list = []
         for button_name, button_link in buttons.items():
-            self.msg.addLinkButton(button_name, button_link)
+            button = {"type": "Action.OpenUrl", "title": button_name, "url": button_link}
+            button_list.append(button)
+        self.msg["attachments"][0]["content"]["actions"] = button_list
+
+    def create_message_header(
+        self,
+        title,
+        subtitle=None,
+    ):
+        title_item = {
+            "type": "TextBlock",
+            "weight": "Bolder",
+            "text": title,
+            "wrap": "true",
+            "style": "heading",
+            "size": "Medium",
+        }
+        title_items = [title_item]
+        if subtitle:
+            subtitle_item = {
+                "type": "TextBlock",
+                "spacing": "Small",
+                "text": subtitle,
+                "isSubtle": "true",
+                "wrap": "true",
+                "size": "Small",
+            }
+            title_items.append(subtitle_item)
+
+        message_header = {
+            "type": "ColumnSet",
+            "columns": [
+                {
+                    "type": "Column",
+                    "items": [
+                        {
+                            "type": "Image",
+                            "url": "https://s3.dualstack.us-east-2.amazonaws.com/pythondotorg-assets/media/community/logos/python-logo-only.png",
+                            "altText": "Python",
+                            "style": "person",
+                            "size": "small",
+                            "horizontalAlignment": "Center",
+                        }
+                    ],
+                    "width": "auto",
+                    "verticalContentAlignment": "Center",
+                },
+                {"type": "Column", "items": title_items},
+            ],
+        }
+        self.body.append(message_header)
+
+    def create_simple_message(self, message):
```

Comment:
Can you add type hinting and docstrings? also for the other functions that are missing these